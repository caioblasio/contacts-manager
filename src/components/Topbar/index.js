import React from "react";
import { Link } from "react-router-dom";
import AppBar from "@material-ui/core/AppBar";
import Toolbar from "@material-ui/core/Toolbar";
import Typography from "@material-ui/core/Typography";

const Topbar = () => (
  <AppBar position="static" color="primary">
    <Toolbar>
      <Link to="/" style={{ textDecoration: "none", color: "inherit" }}>
        <Typography variant="h6">Contacts Manager</Typography>
      </Link>
    </Toolbar>
  </AppBar>
);

export default Topbar;
