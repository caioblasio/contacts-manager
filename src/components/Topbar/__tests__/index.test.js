import React from "react";
import { render } from "@testing-library/react";
import { MemoryRouter } from "react-router";
import Topbar from "components/Topbar";

describe("Given <Topbar />", () => {
  let component;
  let Component;

  beforeAll(() => {
    Component = props => (
      <MemoryRouter>
        <Topbar {...props} />
      </MemoryRouter>
    );
  });

  describe("When it initializes", () => {
    beforeEach(() => {
      const { container } = render(<Component />);
      component = container;
    });

    it("Then should match the snapshot", () => {
      expect(component).toMatchSnapshot();
    });
  });
});
