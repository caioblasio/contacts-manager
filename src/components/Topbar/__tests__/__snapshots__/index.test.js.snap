// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given <Topbar /> When it initializes Then should match the snapshot 1`] = `
<div>
  <header
    class="MuiPaper-root MuiAppBar-root MuiAppBar-positionStatic MuiAppBar-colorPrimary MuiPaper-elevation4"
  >
    <div
      class="MuiToolbar-root MuiToolbar-regular MuiToolbar-gutters"
    >
      <a
        href="/"
        style="text-decoration: none;"
      >
        <h6
          class="MuiTypography-root MuiTypography-h6"
        >
          Contacts Manager
        </h6>
      </a>
    </div>
  </header>
</div>
`;
